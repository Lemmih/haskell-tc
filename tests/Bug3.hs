{-# LANGUAGE RankNTypes #-}
module Bug3 where

higher :: (forall a. a -> a) -> ()
higher id = id ()

bug :: (a -> a) -> ()
bug id = higher id
