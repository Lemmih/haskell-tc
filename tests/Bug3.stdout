UnificationError "Unexpected ref: skolem(a2)"